sys/compat/freebsd64/freebsd64_abort2.c

```diff
@@ -42,7 +44,8 @@ freebsd64_abort2(struct thread *td, struct freebsd64_abort2_args *uap)
 {
 	void * __capability uargs[16];
 	void *uargsp;
-	long ptr;
+	uint64_t * __capability uargscap;
+	uint64_t ptr;
```

Comment:
This should stay a long? `int  fueword_c(volatile const void * __capability base, long *val);`. Doesn't particularly matter, the implicit cast is fine provided `uint64_t` is an `unsigned long` rather than an `unsigned long long`, which it will be on any freebsd64-supporting arch, but there's no need to change this and go against the prototype?

Reply:
In most of the in-kernel compat layers, we try to use fixed-width types rather than native ABI types when holding compat ABI values.  For example, freebsd32 uses `uint32_t` all over (by necessity) instead of `long`.  The Linux compat layers have some extra indirection with `l_long`, etc. types that get mapped to the appropriate fixed-size type for linux32 vs linux.  The reason I used uint64_t here is that `ptr` is a freebsd64-long rather than a native long.    That said, that's also an argument for using fueword64_c instead of a plain fueword_c.